Ctdc 860 Adding file transformation script

wfy1997 opened the pull request.

wfy1997 committed:
Adding transformation scripts

wfy1997 committed:
replace placeholder with .gitkeep

wfy1997 committed:
(1) snv_variant.csv: columns transcript, protein, exon and function replace empty strings with dash -
(2) case.csv: column races , replace AMERICAN_INDIAN_OR_ALASKA_NATIVE, WHITE with Multirace

wfy1997 committed:
Commenting out the rounding values code

wfy1997 committed:
Do not replace empty exon value to “-” in snv.tsv

wfy1997 committed:
replace value "AMERICAN_INDIAN_OR_ALASKA_NATIVE, WHITE" with "Multirace" in "race" column from "case"

wfy1997 committed:
Re-index the transformed data and make "grant_type" and "scope" read from secrets

wfy1997 committed:
Change "True" in the "show_node" column to all cap

wfy1997 committed:
Column "meddra_code" in "case" needs to be int

wfy1997 committed:
Add function to remove trailing ".0" in some data files

wfy1997 committed:
Remove the 'nan' value

wfy1997 committed:
fix the rounding to include one more place up to the 1/100s

wfy1997 committed:
If a clinical_trial.clinical_trial_id column exists in raw file, then copy that column, otherwise add a hard coded column

n2iw reviewed (changes requested):
Suggestions:

- make input folder and output folder configurable
- make input file names configurable, meaning I can config "arm" to be read from "arm-1.csv", output file name should match input file name (except extension)

wfy1997 committed:
make input folder and output folder configurable
make input file names configurable, meaning I can config "arm" to be read from "arm-1.csv", output file name should match input file name (except extension)

wfy1997 committed:
use os.path.join instead of string concatenation, otherwise the leading or trailing “/” will break the code
use os.path.basename instead of str.split(), otherwise if file contain “.” in the name will break the code
please make sure transformation and extraction can use their own config files. Combined config is fine, but separate config files should work also.

wfy1997 committed:
transformation and extraction can use their own config files

wfy1997 committed:
Update to accept transformation config file

Commit:
Added transformation config file and ignored local-config folder.
